engine/protocol/api.go

```diff
@@ -0,0 +1,142 @@
+package protocol
+
+import (
+	"context"
+	"errors"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type API interface {
+	GetLatestBlockHeader(ctx context.Context, isSealed bool) (*flow.Header, error)
+	GetBlockHeaderByID(ctx context.Context, id flow.Identifier) (*flow.Header, error)
+	GetBlockHeaderByHeight(ctx context.Context, height uint64) (*flow.Header, error)
+	GetLatestBlock(ctx context.Context, isSealed bool) (*flow.Block, error)
+	GetBlockByID(ctx context.Context, id flow.Identifier) (*flow.Block, error)
+	GetBlockByHeight(ctx context.Context, height uint64) (*flow.Block, error)
+}
+
+type backend struct {
+	blocks  storage.Blocks
+	headers storage.Headers
+	state   protocol.State
+}
+
+func New(
+	state protocol.State,
+	blocks storage.Blocks,
+	headers storage.Headers,
+) API {
+	b := &backend{
+
+		headers: headers,
+		blocks:  blocks,
+		state:   state,
+	}
+	return b
+}
+
```

Comment:
Since this will be used as a standalone API, perhaps we should include a Ping route for healthcheck. See access/api for example.